engine/lib/output.php

```diff
@@ -142,6 +142,11 @@ function elgg_format_attributes(array $attrs = array()) {
 	$attributes = [];
 
 	foreach ($attrs as $attr => $val) {
+		if (!is_string($attr) || $attr[0] == '#') {
```

Comment:
you could wonder why the hashed attribute is even added.... this is fixing symptoms.. if you want to do that, we need a better check for invalid attribute_names (not only starting with a hash)